libnopegl/src/pipeline_compat.c

```diff
@@ -401,6 +401,58 @@ int ngli_pipeline_compat_update_buffer(struct pipeline_compat *s, int32_t index,
     return 0;
 }
 
+static int select_next_available_bindgroup(struct pipeline_compat *s)
+{
+    struct gpu_ctx *gpu_ctx = s->gpu_ctx;
+
+    s->cur_bindgroup = s->bindgroups[s->next_bindgroup++];
+    s->next_bindgroup %= s->nb_bindgroups;
+
+    /* If the selected bindgroup is not in used by any command buffer, select it */
+    if (s->cur_bindgroup->rc.count == 1)
+        return 0;
+
+    /* Otherwise, increase our bindgroup pool */
+    size_t nb = s->nb_bindgroups;
+    size_t nb_bindgroups = s->nb_bindgroups * 2;
+    struct bindgroup **bindgroups = ngli_realloc(s->bindgroups, nb_bindgroups, sizeof(*bindgroups));
+    if (!bindgroups)
+        return NGL_ERROR_MEMORY;
+
+    memset(bindgroups + s->nb_bindgroups, 0, s->nb_bindgroups * sizeof(*bindgroups));
```

Comment:
`darray`?

Reply:
I'm currently not using any darray around here so I would rather keep it as is.

Reply:
Changed to darray and simplified the logic.